Documentation/MigrationGuide.md

```diff
@@ -0,0 +1,60 @@
+
+Instead, you're now **required** to provide a template when invoking SwiftGen, either via `-t <templatename>` (you can see the list of available template names using the `swiftgen templates list` command) or via `-p <templatepath>`.
+
+ℹ️ The templates that were used as default template in SwiftGen 4.0 are still bundled with SwiftGen, but they have been renamed (mainly to `"swift2"` since the "default" template in SwiftGen 4 was the one we wrote back in Swift 2 times!) and you now have to explicitly tell which to use. See [below](#templates-bundled-with-swiftgen--some-renaming) for more info.
+
+✅ **Most of you probably already use the `-t swift3` option** if you were writing Swift3 code, so **you won't need to change anything**. But in the unlikely event that you were still writing Swift2 and didn't specify a template, you'll then have to use `-t swift2` to force using this (previously default) template.
+
+## Bundled Templates have been renamed & cleaned
+
+Some templates bundled with SwiftGen have been **renamed** to have a better naming consistency.
+
+Be sure to consult [the templates repo's own Migration Guide](https://github.com/SwiftGen/templates/blob/master/Documentation/MigrationGuide.md) to see the list of changes (renamings or removal) in templates bundled in SwiftGen.
+
+📖 Also, **each template bundled in SwiftGen has a dedicated documentation now**, listing what the template is for, what does the generated code looks like, when you might want to choose that template over another one, and the parameters (`--param X=Y`) supported by this template for customization via the command line.  
+This should help you choose the right template to use for your use case (or help you decide if you need to create your own if not of the provided ones fit your needs) and see what's customizable for each.
+
+📖 See [the documentation folder in the templates repo](https://github.com/SwiftGen/templates/tree/master/Documentation) for those templates documentation. This folder is organized the same way the templates are: one subfolder for each SwiftGen command (`colors`, `strings`, …), then one markdown file for each eponym template.
+
+## If you wrote your own templates
+
+### Stencil Context keys have been refactored
+
+If you decided to write your own templates for SwiftGen 4, you'll have to amend them to fit the new names for context variables, as some of the variables provided by SwiftGenKit to your templates have been renamed for more consistency.
+
+For example:
+
+* now that the `colors` command supports more than one color palette, the context's root key `colors` that you used to iterate over the list of colors has been replaced by the root key `palettes` listing all the palettes parsed by SwiftGen, with their `name` and `colors`. Also `rgb` and `rgba` has been deprecated—they can be recreated from the `red`, `green`, `blue` keys.
```

Comment:
Adding some surrounding spaces around `—` in `...deprecated — they...` would read better (like the one just below).